yb_build_thirdparty_main.py

```diff
@@ -958,6 +958,87 @@ def get_openssl_related_cmake_args(self):
         ]
         return openssl_options
 
+class LibTest:
```

Comment:
I think this class is better split into a hierarchy of a base class and Linux / macOS specific checkers.